### PR TITLE
Decode HTML entities for title in social sharing component

### PR DESCRIPTION
**Pull request by amyevans**

### CODE REVIEW

File: tests/components/test-components.php [hunk 1]
```diff
@@ -883,7 +883,7 @@ public function test_component_post_social_sharing() {
 					],
 					'platformData' => [],
 					'postId'       => $this->get_post_id(),
-					'title'        => get_the_title( $this->get_post_id() ),
+					'title'        => 'It’s like this & that.',
```
Comment (joemcgill) [inline tests/components/test-components.php:886]: Do we need to update the `get_the_title()` helper to make encoding optional?

**Review by joemcgill — APPROVED** [review]
Are there cases where the title property is used where it _should_ be encoded? If not, then I think this is fine. 👍🏻

File: tests/components/test-components.php (hunk 1 above)
Comment (amyevans) [reply tests/components/test-components.php:886]: Hmm, there isn't an existing helper, we're just using the core function. I can add a helper though.

**Comment by amyevans** [comment]
@joemcgill I added a few more commits - could you take another look?

**Review by joemcgill — APPROVED** [review]
👍🏻 changes look good to me

**Pull request merged**